### PR TITLE
Task-37846 : Active DLP restore action

### PR DESCRIPTION
**Pull request by ahmadmido**

### CODE REVIEW

File: webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties [hunk 1]
```diff
@@ -8,3 +8,8 @@ documents.dlp.quarantine.author=Author
 documents.dlp.quarantine.previewDownload=Preview / download the document
 documents.dlp.quarantine.validateDoc=Validate the document
 documents.dlp.quarantine.deleteDoc=Delete the document
+
+documents.dlp.message.confirmRestore=Are you sure you want to restore this document?
+documents.dlp.title.confirmRestore=Restore document
+documents.dlp.button.ok=OK
+documents.dlp.button.cancel=Cancel
```
Comment (rdenarie) [inline webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties]: this should be modified to stay generic.
Documents is KO. It should be Item or Element

File: webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue [hunk 2]
```diff
@@ -152,6 +162,15 @@ export default {
         this.dlpFeatureStatusLoaded = true;
       });
     },
+    restoreDlpPositiveDocumentConfirm() {
+      dlpAdministrationServices.restoreDlpPositiveItem(this.selectedDocId);
+      this.retrieveDlpPositiveItems();
+    },
+    restoreDlpPositiveItem(docId) {
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue]:
```suggestion
    restoreDlpPositiveItem(itemId) {
```

File: webapp/portlet/src/main/webapp/dlp-quarantine/dlpAdministrationServices.js [hunk 3]
```diff
@@ -37,4 +37,14 @@ export function getDlpPositiveItems(offset, itemsPerPage) {
       throw new Error('Error retrieving list of dlp positive items');
     }
   });
-} 
+}
+
+export function restoreDlpPositiveItem(docId) {
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/dlp-quarantine/dlpAdministrationServices.js]:
```suggestion
export function restoreDlpPositiveItem(itemId) {
```

File: webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue [hunk 4]
```diff
@@ -152,6 +162,15 @@ export default {
         this.dlpFeatureStatusLoaded = true;
       });
     },
+    restoreDlpPositiveDocumentConfirm() {
+      dlpAdministrationServices.restoreDlpPositiveItem(this.selectedDocId);
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue]: remove the reference to document

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties [hunk 5]
```diff
@@ -5,11 +5,14 @@ documents.dlp.quarantine.keywordDetected=Keyword Detected
...
-documents.dlp.button.cancel=Cancel
-documents.dlp.error.UnknownServerError=An unknown error happened when processing operation. Please contact your administrator.
+
+items.dlp.quarantine.previewDownload=Preview / download the document
+items.dlp.quarantine.validate=Validate the document
+items.dlp.quarantine.delete=Delete the document
+items.dlp.message.confirmRestore=Are you sure you want to restore this document?
+items.dlp.title.confirmRestore=Restore document
+items.dlp.message.confirmDelete=Are you sure you want to delete this document?
+items.dlp.title.confirmDelete=Delete document
```
Comment (rdenarie) [inline webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties]:
```suggestion
items.dlp.quarantine.previewDownload=Preview / download this item
items.dlp.quarantine.validate=Validate item
items.dlp.quarantine.delete=Delete item
items.dlp.message.confirmRestore=Are you sure you want to restore this item?
items.dlp.title.confirmRestore=Restore item
items.dlp.message.confirmDelete=Are you sure you want to delete this item?
items.dlp.title.confirmDelete=Delete item
```

File: webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue [hunk 6]
```diff
@@ -183,17 +193,26 @@ export default {
         }, 5000);
       }).finally(() => this.loading = false);
     },
-    deleteDlpPositiveDocument(documentId) {
-      this.deleteConfirmMessage = this.$t('documents.dlp.message.confirmDelete');
+    deleteDlpPositiveDocument(itemId) {
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue]:
```suggestion
    deleteDlpPositiveItem(itemId) {
```

File: webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue [hunk 7]
```diff
@@ -183,17 +193,26 @@ export default {
...
-      this.selectedItem = documentId;
+      this.selectedDeleteItem = itemId;
     },
     getDlpFeatureStatus() {
       dlpAdministrationServices.isDlpFeatureActive().then(status => {
         this.dlpFeatureEnabled = status.value;
         this.dlpFeatureStatusLoaded = true;
       });
     },
+    restoreDlpPositiveDocumentConfirm() {
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/dlp-quarantine/components/DlpQuarantineApp.vue]:
```suggestion
    restoreDlpPositiveItemConfirm() {
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties (hunk 5 above)
Comment (ahmadmido) [reply webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties]: Ce est mis dans une autre nouvelle tache.

Comment (rdenarie) [reply webapp/portlet/src/main/resources/locale/portlet/dlpQuarantine/dlpQuarantine_en.properties]: Here you modify this keys. So we can modify the value in this PR

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**